homeassistant/components/tuya/light.py

```diff
@@ -0,0 +1,381 @@
+
+        if ATTR_HS_COLOR in kwargs:
+            colour_data = self._get_hsv()
+            # hsv h
+            colour_data["h"] = int(kwargs[ATTR_HS_COLOR][0])
+            # hsv s
+            ha_s = kwargs[ATTR_HS_COLOR][1]
+            s_range = self._tuya_hsv_s_range()
+            colour_data["s"] = int(
+                self.remap(
+                    ha_s,
+                    HSV_HA_SATURATION_MIN,
+                    HSV_HA_SATURATION_MAX,
+                    s_range[0],
+                    s_range[1],
+                )
+            )
+            # hsv v
+            ha_v = self.brightness
+            v_range = self._tuya_hsv_v_range()
+            colour_data["v"] = int(self.remap(ha_v, 0, 255, v_range[0], v_range[1]))
+
+            commands += [
+                {"code": self.dp_code_colour, "value": json.dumps(colour_data)}
+            ]
+            if self.tuya_device.status[DPCODE_WORK_MODE] != "colour":
+                commands += [{"code": DPCODE_WORK_MODE, "value": "colour"}]
+
+        if ATTR_COLOR_TEMP in kwargs:
+            # temp color
+            new_range = self._tuya_temp_range()
+            color_temp = self.remap(
+                self.max_mireds - kwargs[ATTR_COLOR_TEMP] + self.min_mireds,
+                self.min_mireds,
+                self.max_mireds,
+                new_range[0],
+                new_range[1],
+            )
+            commands += [{"code": self.dp_code_temp, "value": int(color_temp)}]
+
+            # brightness
+            ha_brightness = self.brightness
+            new_range = self._tuya_brightness_range()
+            tuya_brightness = self.remap(
+                ha_brightness, 0, 255, new_range[0], new_range[1]
+            )
+            commands += [{"code": self.dp_code_bright, "value": int(tuya_brightness)}]
+
+            if self.tuya_device.status[DPCODE_WORK_MODE] != "white":
+                commands += [{"code": DPCODE_WORK_MODE, "value": "white"}]
+
+        self._send_command(commands)
+
+    def turn_off(self, **kwargs: Any) -> None:
+        """Instruct the light to turn off."""
+        if (
+            DPCODE_LIGHT in self.tuya_device.status
+            and DPCODE_SWITCH not in self.tuya_device.status
+        ):
+            commands = [{"code": DPCODE_LIGHT, "value": False}]
+        else:
+            commands = [{"code": DPCODE_SWITCH, "value": False}]
+        self._send_command(commands)
+
+    @property
+    def brightness(self):
+        """Return the brightness of the light."""
+        old_range = self._tuya_brightness_range()
+        brightness = self.tuya_device.status.get(self.dp_code_bright, 0)
+
+        _LOGGER.debug(
+            "brightness id-> %s, work_mode-> %s, check true-> %s", 
+            self.tuya_device.id, 
+            self._work_mode(), 
+            self._work_mode().startswith(WORK_MODE_COLOUR)
+        )
+
+        if self._work_mode().startswith(WORK_MODE_COLOUR):
+            colour_data = json.loads(
```

Comment:
Why doesn't the library provide a usable format from the beginning from its interface?

Reply:
This is a good suggestion, we will evaluate it internally